control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -609,9 +609,15 @@ func (r *reconciler) reconcileKonnectivityAgent(ctx context.Context, hcp *hyperv
 		}
 	}
 
+	proxy := &configv1.Proxy{ObjectMeta: metav1.ObjectMeta{Name: "cluster"}}
+	if err := r.client.Get(ctx, client.ObjectKeyFromObject(proxy), proxy); err != nil {
+		// If the cluster doesn't use a proxy this is irrelevant, so we don't return.
```

Comment:
should we not retry if the error is something other than NotFound?

Reply:
We will retry because the error gets end up being returned, we will just not block the DS creation with this

Reply:
ack